src/main/python/ayab/utils.py

```diff
@@ -64,38 +71,39 @@ def populate_ports(combo_box=None, port_list=None):
         combo_box.addItem(item[0])
 
 
-def get_serial_ports():
+def get_serial_ports()->list[str]:
     """Return a list of all USB serial ports."""
-    return list(serial.tools.list_ports.grep("USB"))
+    return list(serial.tools.list_ports.grep("USB")) #type: ignore
```

Comment:
`.grep` returns a `Generator` and `list` constructor doesn't like that for some reason. I don't get why, it's a typing thing not a code issue.